### PR TITLE
Alert dialog 2

### LINKED ISSUE #60: Alert Dialog
Screen Shot 2021-05-02 at 10 38 08 PM
Screen Shot 2021-05-02 at 10 38 16 PM

Progress can be tracked [here](https://github.com/GroovinChip/macos_ui/tree/alert_dialog).

### PR DESCRIPTION
**Pull request by GroovinChip**
This PR contains version 0.5.0, which adds the `MacosAlertDialog` widget

Closes #60 

## Pre-launch Checklist

- [x] I have run `dartfmt` on all changed files 
- [x] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes 
- [x] I have added/updated relevant documentation 
- [x] I have run "optimize/organize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could

### CODE REVIEW

File: lib/src/dialogs/macos_alert_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            const SizedBox(height: 28),
+            ConstrainedBox(
+              constraints: BoxConstraints(
+                maxHeight: 56,
+                maxWidth: 56,
+              ),
+              child: appIcon,
+            ),
+            const SizedBox(height: 28),
+            title,
```
Comment (bdlukaa) [inline lib/src/dialogs/macos_alert_dialog.dart]: Wrap this in a `DefaultTextStyle` and set `MacosTheme.of(context).typography.headline` to `style`

File: example/lib/pages/dialogs_page.dart [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+          child: Text('Show Alert Dialog 2'),
+          onPressed: () => showDialog(
+            context: context,
+            builder: (_) => MacosAlertDialog(
+              appIcon: FlutterLogo(
+                size: 56,
+              ),
+              title: Text(
+                'Alert Dialog with Secondary Action',
+                style: MacosTheme.of(context).typography.headline,
```
Comment (bdlukaa) [inline example/lib/pages/dialogs_page.dart]: the text style shouldn't be provided here

File: example/lib/pages/dialogs_page.dart [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+              ),
+              message: Text(
+                'This is an alert dialog with primary action and secondary action laid out horizontally',
+                textAlign: TextAlign.center,
+                style: MacosTheme.of(context).typography.headline,
```
Comment (bdlukaa) [inline example/lib/pages/dialogs_page.dart]: These should not be provided here

File: lib/src/dialogs/macos_alert_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            const SizedBox(height: 16),
+            Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 16.0),
+              child: message,
```
Comment (bdlukaa) [inline lib/src/dialogs/macos_alert_dialog.dart]: Wrap `message` in a `DefaultTextStyle` and set
```dart
textAlign: TextAlign.center,
style: MacosTheme.of(context).typography.headline,
```

File: lib/src/dialogs/macos_alert_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    required this.primaryButton,
+    this.secondaryButton,
+    this.horizontalActions = true,
+  }) : super(key: key);
+
+  /// This should be your application's icon.
+  ///
+  /// The size of this widget should be 56x56.
+  final Widget appIcon;
+  final Widget title;
```
Comment (bdlukaa) [inline lib/src/dialogs/macos_alert_dialog.dart:28]: Documentation.

`Typically a Text widget`

File: lib/src/dialogs/macos_alert_dialog.dart [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+  final Widget message;
```
Comment (bdlukaa) [inline lib/src/dialogs/macos_alert_dialog.dart:33]: documentation

`Typically a Text widget`

File: lib/src/dialogs/macos_alert_dialog.dart [hunk 7]
```diff
@@ -0,0 +1,136 @@
+import 'package:flutter/material.dart';
+import 'package:macos_ui/macos_ui.dart';
+
+/// A macOS-style AlertDialog.
+///
+/// A [MacosAlertDialog] must display an [appIcon], [title], [message],
+/// and [primaryButton].
+class MacosAlertDialog extends StatelessWidget {
+  const MacosAlertDialog({
```
Comment (bdlukaa) [inline lib/src/dialogs/macos_alert_dialog.dart:10]: Documentation on the constructor

**Review by bdlukaa — CHANGES_REQUESTED** [review]
I can see you're using the `showDialog` function from the material library. I think the function should be implemented on the `macos_ui` library

**Comment by GroovinChip** [comment]
> I can see you're using the `showDialog` function from the material library. I think the function should be implemented on the `macos_ui` library

Why reinvent the wheel? It works, and works well.

**Review by bdlukaa — APPROVED** [review]
LGTM

**Pull request merged**